### DISCUSSION TITLE
How Do I Get Recycle Bin List Id via PnP

### DISCUSSION THREAD

**Discussion by YuriySamorodov** [pnp/powershell#226] [Q&A]
Just wondering if I can get Recycle Bin list id via PnP somehow. An idea is to get to a list/library settings page to review item count like I can do with otherlibraries: `/_layouts/15/listedit.aspx?List={4017a0b5-0ce2-4fdb-9538-ec96a5f6e977}`

**Comment by LeonArmston** [comment]
I'm not aware of an easy  way to get the list ID GUID for the recycle bin - you can get the recycle bin item count for a site using 

((Get-PnPRecycleBinItem).Count)

**Reply by YuriySamorodov** [reply]
Thanks @LeonArmston.
Unfortunately `Get-PnPRecycleBinItem` will error out if there are over 5000 items...

**Reply by LeonArmston** [reply]
Hi Yuri

Get-PnPRecycleBinItem & Clear-PnpRecycleBinItem now have -RowLimit parameters that you could use to retrieve less than 5000 items.

Looks like this issue has cropped up before in June last year and you may find this [issue thread](https://github.com/pnp/PnP-PowerShell/issues/2315) useful to read.

Thanks

**Reply by YuriySamorodov** [reply]
Thanks for the heads up @LeonArmston That's really helpful.
I was hoping there  was a faster way rather than getting item count with loops.
